### ISSUE TITLE
The shell integration appears on the wrong line when using the Hydro prompt

### ISSUE THREAD

**Issue by mattmc3** [kidonng/vscode.fish#1]
Using the popular Hydro prompt (https://github.com/jorgebucaran/hydro), the VSCode shell integration appears on the line below the prompt instead of the actual prompt line:

<img width="526" alt="Screen Shot 2022-08-08 at 6 39 47 PM" src="https://user-images.githubusercontent.com/978625/183526749-6d1650c2-1bb6-4891-af99-053e04cb9cc0.png">

**Comment by kidonng** [comment]
Ah, async prompt. I don't think there is out of box solution for them, maybe we can try provide a general hook and wrap whatever function needed to show the prompt.

**Comment by mattmc3** [comment]
Makes sense. If we can determine what that would need to look like, I’m willing to help put together a small PR for either project.

@jorgebucaran - FYI for visibility
